modules/oe_content_event/tests/src/Kernel/EventLinkSourceFilterTest.php

```diff
@@ -0,0 +1,167 @@
+        'end_value' => '2050-05-15T12:00:00',
+      ],
+    ];
+    $upcoming_event = Node::create($values);
+    $upcoming_event->save();
+
+    // Create an event that starts on the far future.
+    $values = [
+      'type' => 'oe_event',
+      'title' => 'My node title',
+      'oe_event_dates' => [
+        'value' => '2200-05-10T12:00:00',
+        'end_value' => '2200-05-15T12:00:00',
+      ],
+    ];
+    $future_event = Node::create($values);
+    $future_event->save();
+
+    /** @var \Drupal\oe_link_lists_internal_source\InternalLinkSourceFilterPluginManager $plugin_manager */
+    $plugin_manager = $this->container->get('plugin.manager.oe_link_lists.internal_source_filter');
+    /** @var \Drupal\oe_link_lists_internal_source\InternalLinkSourceFilterInterface $plugin */
+    $plugin = $plugin_manager->createInstance('oe_content_event_period', []);
+
+    // Plugin applies to events.
+    $plugins = $plugin_manager->getApplicablePlugins('node', 'oe_event');
+    $this->assertTrue(reset($plugins) instanceof EventPeriodFilter);
+    // Plugin does not apply to news.
+    $plugins = $plugin_manager->getApplicablePlugins('node', 'oe_news');
+    $this->assertEmpty($plugins);
+
+    $cache = new CacheableMetadata();
+    /** @var \Drupal\node\NodeStorage $storage */
+    $storage = $this->container->get('entity_type.manager')->getStorage('node');
+
+    // Default filter will only return upcoming events.
+    $query = $storage->getQuery();
+    $plugin->apply($query, [], $cache);
+    $query_results = $query->execute();
+    $future_events = [
+      $upcoming_event->id() => $upcoming_event->id(),
+      $future_event->id() => $future_event->id(),
+    ];
+    $this->assertEqual($query_results, $future_events);
+    // Time based caches have been added and they correspond to the closest
+    // upcoming event start date.
+    $date_cache_tags = [
+      'oe_time_caching_date:2050',
+      'oe_time_caching_date:2050-05',
+      'oe_time_caching_date:2050-05-10',
+      'oe_time_caching_date:2050-05-10-12',
+    ];
+    $this->assertEqual($cache->getCacheTags(), $date_cache_tags);
+
+    // Configuring the filter for past events will only return finished events.
+    $query = $storage->getQuery();
+    $plugin->setConfiguration(['period' => EventPeriodFilter::PAST]);
+    $cache = new CacheableMetadata();
+    $plugin->apply($query, [], $cache);
+    $query_results = $query->execute();
+    $past_events = [
+      $past_event->id() => $past_event->id(),
+      $ancient_event->id() => $ancient_event->id(),
+    ];
+    $this->assertEqual($query_results, $past_events);
+    // Time based caches have been added and they correspond to the closest
+    // event end date.
+    $date_cache_tags = [
+      'oe_time_caching_date:2050',
+      'oe_time_caching_date:2050-05',
+      'oe_time_caching_date:2050-05-15',
+      'oe_time_caching_date:2050-05-15-12',
+    ];
+    $this->assertEqual($cache->getCacheTags(), $date_cache_tags);
```

Comment:
These assertions are exactly the same as the future one. We can have different generated tags by setting the current time between an event's start and end date? So we fall into this case:
```php
      case self::PAST:
        $query->condition('oe_event_dates.end_value', $now->format(DateTimeItemInterface::DATETIME_STORAGE_FORMAT), "<");
        $query->sort('oe_event_dates.end_value', 'DESC');
```

Reply:
Actually we are already checking that that: the tags correspond to the event's end date (the day is different!)

Reply:
True that, my bad.